mobile/status_request_log.go

```diff
@@ -0,0 +1,51 @@
+package statusgo
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/status-im/status-go/logutils"
+)
+
+// logAndCall logs request call details and executes the fn function if logging is enabled
+func logAndCall[T any](method string, params any, fn func() (T, error)) (T, error) {
+	var startTime time.Time
+	var duration time.Duration
+
+	if logutils.IsRequestLoggingEnabled() {
+		startTime = time.Now()
+	}
+
+	resp, err := fn()
+
+	if logutils.IsRequestLoggingEnabled() {
+		duration = time.Since(startTime)
+		logutils.GetRequestLogger().Debug(method, "params", fmt.Sprintf("%+v", params), "resp", fmt.Sprintf("%+v", resp), "err", err, "duration", duration)
+	}
+
+	return resp, err
+}
+
+func logAndCallString(method string, params any, fn func() string) string {
+	resp, _ := logAndCall(method, params, func() (string, error) {
+		return fn(), nil
+	})
+	return resp
+}
+
+func logAndCallVoid(method string, params any, fn func()) {
+	_, _ = logAndCall(method, params, func() (any, error) {
+		fn()
+		return nil, nil
+	})
+}
+
+func removeSensitiveInfo(jsonStr string) string {
+	// see related test for the usage of this function
+	re := regexp.MustCompile(`(?i)(".*?(password|mnemonic).*?")\s*:\s*("[^"]*")`)
+	return re.ReplaceAllStringFunc(jsonStr, func(match string) string {
+		parts := re.FindStringSubmatch(match)
+		return fmt.Sprintf(`%s:"[REDACTED]"`, parts[1])
```

Comment:
I think `***` is usually used in such cases.
```suggestion
		return fmt.Sprintf(`%s:"***"`, parts[1])
```

Reply:
fixed